demo/contacts/contact-data-source.mjs

```diff
@@ -0,0 +1,43 @@
+import * as util from '../util/util.mjs';
+
+// All loads after the first will be delayed by this much.
+const DELAY_MS = 1000;
+
+export class ContactDataSource {
+  #delayMs = 0;
+  #allContacts = null;
+  #loading = false;
+  #loadedAll = false;
+
+  get loadedAll() {
+    return this.#loadedAll;
+  }
+
+  get loading() {
+    return this.#loading;
+  }
+
+  get delay() {
+    return this.#delayMs;
+  }
+
+  set delay(delayMs) {
+    this.#delayMs = delayMs;
+  }
+
+  async getNextContacts(count) {
+    this.#loading = true;
+    if (!this.#allContacts) {
+      this.#allContacts =
+          await fetch('contacts/contacts.json').then(resp => resp.json());
+    }
+    // Simulate slow server load...
+    await util.delay(this.#delayMs);
+    const res = this.#allContacts.splice(0, count);
+    this.#loadedAll = this.#allContacts.length === 0;
+    this.#loading = false;
+    // After the first load, simulate a delay.
+    this.#delayMs = DELAY_MS;
```

Comment:
Having the public setter seems weird if you're going to reset the value every time getNextContacts is called. Maybe either:

- Remove the public setter/getter
- Have a `#firstLoad` private boolean which lets you ignore `this.#delayMs` for the first load.

Reply:
The only demo that used it was fallback.html and I don't see a reason to port that demo over. So I've just deleted them.